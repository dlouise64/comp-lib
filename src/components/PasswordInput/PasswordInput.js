import React from 'react';
import PropTypes from 'prop-types';
import ProgressBar from '../ProgressBar';
import EyeIcon from '../EyeIcon';
import TextInput from '../TextInput';

// import './PasswordInput.css';

class PasswordInput extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      showPassword: false
    }
  }

  toggleShowPassword = event => {
    this.setState(prevState => {
      return { showPassword: !prevState.showPassword };
    });
    if (event) event.preventDefault();
  }
  render() {
    const { htmlId, value, label, error, onChange, placeholder, maxLength, showVisibilityToggle, quality, ...props} = this.props;
    const { showPassword } = this.state;

    return (
      <TextInput
        htmlId={htmlId}
        label={label}
        placeholder={placeholder}
        type={showPassword ? 'text' : 'password'}
        onChange={onChange}
        value={value}
        maxLength={maxLength}
        error={error}
        required
        {...props}>
        {
          showVisibilityToggle &&
        <span
          onClick={this.toggleShowPassword}
          style={{
            position: 'absolute',
            right: '-35px',
            top: '27px',
            bottom: 0
          }}
        >
          <EyeIcon />
        </span>
        }
        {
          value.length > 0 && quality && <ProgressBar percent={quality} width={206} height={7} />
        }
      </TextInput>
    );
  }
}

PasswordInput.propTypes = {
  /** Unique HTML ID */
  htmlId: PropTypes.string.isRequired,

  /** Input name */
  name: PropTypes.string.isRequired,

  /** Password value */
  value: PropTypes.any,

  /** Input label */
  label: PropTypes.string,

  /** Function called when password input value changes */
  onChange: PropTypes.func.isRequired,

  /** Max password length accepted */
  maxLength: PropTypes.number,

  /** Placeholder */
  placeholder: PropTypes.string,

  /** set to true to show the toggle for displaying the currently entered password */
  showVisibilityToggle: PropTypes.bool,

  /** Display password quality visually via ProgressBar, accepts a number between 0 and 100 */
  quality: PropTypes.number,

  /** Validation error to display */
  error: PropTypes.string
};

PasswordInput.defaultProps = {
  maxLength: 50,
  showVisibilityToggle: false,
  label: 'Password'
}

export default PasswordInput;
