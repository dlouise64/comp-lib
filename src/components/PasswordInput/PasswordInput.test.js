import React from 'react';
import renderer from 'react-test-renderer';
import PasswordInput from './PasswordInput';

test('<PasswordInput /> renders correctly', () => {
  const tree = renderer.create(<PasswordInput />).toJSON();
  expect(tree).toMatchSnapshot();
});
