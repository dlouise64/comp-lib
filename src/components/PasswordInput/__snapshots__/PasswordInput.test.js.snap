// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordInput /> renders correctly 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
      "marginBottom": 16,
      "position": "relative",
    }
  }
>
  <label
    className="label"
    htmlFor="password"
    style={
      Object {
        "display": "inline-block",
        "marginRight": "10px",
      }
    }
  >
    Password
     
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </label>
  <input
    className="textInput"
    id="password"
    maxLength={50}
    minLength={8}
    name="password"
    onChange={[Function]}
    placeholder="Enter Password"
    type="password"
    value=""
    width={120}
  />
  <span
    onClick={[Function]}
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "right": "-35px",
        "top": "27px",
      }
    }
  >
    <svg
      height="24"
      viewBox="0 0 22 22"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="#4d4d4d"
        transform="matrix(.02146 0 0 .02146 1 1)"
      >
        <path
          d="m466.07 161.53c-205.6 0-382.8 121.2-464.2 296.1-2.5 5.3-2.5 11.5 0 16.9 81.4 174.9 258.6 296.1 464.2 296.1 205.6 0 382.8-121.2 464.2-296.1 2.5-5.3 2.5-11.5 0-16.9-81.4-174.9-258.6-296.1-464.2-296.1m0 514.7c-116.1 0-210.1-94.1-210.1-210.1 0-116.1 94.1-210.1 210.1-210.1 116.1 0 210.1 94.1 210.1 210.1 0 116-94.1 210.1-210.1 210.1"
        />
        <circle
          cx="466.08"
          cy="466.02"
          r="134.5"
        />
      </g>
    </svg>
  </span>
</div>
`;
